### PR TITLE
Iter 2 mb

### PR DESCRIPTION
**Pull request by Maxwell-Baird**
Added percent of home wins, percent of visitor wins, percent of ties, a count of games by season, average goals per game, and average goals by season.
Also added test to check the methods

### CODE REVIEW

File: lib/game_stats.rb [hunk 1]
```diff
@@ -9,6 +9,54 @@ def initialize(file_path, object)
     @games = csv_data('./data/games_truncated.csv', Game)
   end
 
+  def percentage_ties
+    ties = @games.count do |game|
+      game.away_goals == game.home_goals
+    end
+    (100 * ties.fdiv(@games.length)).round(2)
```
Comment (whitneykidd) [inline lib/game_stats.rb:16]: Nice use of .fdiv here

File: lib/game_stats.rb [hunk 2]
```diff
@@ -9,6 +9,54 @@ def initialize(file_path, object)
...
+  end
+
+  def average_goals_by_season
+    goals_per_season = {}
+    @games.each do |game|
+      if goals_per_season[game.season] == nil
+        goals_per_season[game.season] = game.away_goals + game.home_goals
+      else
+        goals_per_season[game.season] += game.away_goals + game.home_goals
+      end
```
Comment (alex-latham) [inline lib/game_stats.rb:51]: Future opportunity to refactor into a reduce enumerable. 
`@games.reduce(Hash.new(0)) { |game| 
    goals_per_season[game.season] += game.away_goals + game.home_goals;
    goals_per_season
}`

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -1,15 +1,18 @@
 require 'csv'
+require './lib/data_module'
```
Comment (tylertomlinson) [inline lib/stat_tracker.rb]: Don't need this here anymore. Also will need to require_relative on our files moving forward

File: test/stat_tracker_test.rb [hunk 4]
```diff
@@ -20,8 +20,9 @@ def test_stat_tracker_exists
   end
 
   def test_it_can_initialize_from_correct_file_plath
-    assert_equal './data/games_truncated.csv', @stat_tracker.game_path
+    assert_equal './data/games_truncated.csv', @stat_tracker.games_path
```
Comment (alex-latham) [inline test/stat_tracker_test.rb:23]: Good catches

File: test/game_stats_test.rb [hunk 5]
```diff
@@ -28,15 +28,48 @@ def test_attributes_for_instance_of_game_withing_game_stats
...
+    assert_equal 57.14, @games_stats.percentage_home_wins
+  end
+
+  def test_it_can_count_games_by_season
+    games_by_season = {
+      "20152016" => 3,
+      "20132014" => 2,
+      "20142015" => 1,
+      "20162017" => 1
+    }
```
Comment (alex-latham) [inline test/game_stats_test.rb:45]: We can refactor this later to match the hash in test_it_can_calculate_average_goals_per_season.

File: lib/stat_tracker.rb (hunk 3 above)
Comment (Maxwell-Baird) [reply lib/stat_tracker.rb]: fixed thanks for pointing it out

**Pull request merged**